libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/FeedbackMetadata.kt

```diff
@@ -0,0 +1,195 @@
+package com.mapbox.navigation.core.telemetry.events
+
+import android.location.Location
+import com.google.gson.Gson
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.internal.utils.toTelemetryLocations
+import com.mapbox.navigation.core.telemetry.LocationsCollector
+
+/**
+ * It's the wrapper of [FeedbackMetadata] that collect locations after instance of
+ * [FeedbackMetadataWrapper] is created and stop collecting in 2 possible cases:
+ *
+ * - locations buffer is full;
+ * - [FeedbackMetadataWrapper.get] is call.
+ *
+ * [FeedbackMetadataWrapper.get] can be invoked whenever it needs.
```

Comment:
Let's be more specific. I see 2 scenarios now, the first when developers want to serialize the object they shouldn't serialize the wrapper but only the resulting metadata, the second is right before posting feedback. Is there anything more that we should highlight?

Reply:
> the second is right before posting feedback.

recommended, but optional

> Is there anything more that we should highlight?

I think that should be enough

Reply:
How could you post feedback without unwrapping metadata, you have to do it, right?

Reply:
current docs + docs of `get()` should be enough:
```
    /**
     * Provide [FeedbackMetadata] with locations that were collected after creating instance of
     * [FeedbackMetadataWrapper].
     */
    fun get(): FeedbackMetadata
```
Do you think we have to add some more?

Reply:
LGTM, just added one comment below but otherwise :ship: